### ISSUE TITLE
Multiline regex support 

### ISSUE THREAD

**Issue by bstaletic** [hanickadot/compile-time-regular-expressions#41]
First of all, thanks for this library.

I did manage to get CTRE to work for a test sample, but just like `std::regex` CTRE doesn't seem to support multiline regex patterns, where `^` and `$` also match line separators.

Example where this would be useful:
```C++
#include <cassert>
#include <iostream>
#include <vector>
#include "include/ctll.hpp"
#include "include/ctre.hpp"

static constexpr auto pattern = ctll::basic_fixed_string{ "^(\\w)\\t(\\w)(?:$|\\n|\\r\\n)" };

int main(void) {
	std::string str{ "a\ta\nc\td" };
	auto start = str.begin();
	auto end = str.end();
	std::vector<std::string> first;
	std::vector<std::string> second;
	while ( auto match = ctre::re<pattern>().search_2( start, end ) ) {
		start = match.get_end_position();
		first.emplace_back( match.get<1>().to_view() );
		second.emplace_back( match.get<2>().to_view() );
	}
	assert( (first == std::vector<std::string>{ "a", "c" }) );
	assert( (second == std::vector<std::string>{ "a", "d" }) );
	return 0;
}
```

The above example works currently, but with multiline regex the last, non-capturing, group could be just `$`. This kind of multiline regex works currently in Boost.Regex and is supposed to work in C++17 `std::regex`, but none of the three most common STL implementations has multiline available.

**Comment by BenHanson** [comment]
+1

**Comment by bstaletic** [comment]
Thank you for looking into this. I know this feature would be quite intrusive. Another useful modifier would be for setting whether `.` matches new line. With Boost.Regex there's `static const match_flag_type match_not_dot_newline;`. `std::regex` doesn't allow `.` to match new line and [pcre documentation](http://www.pcre.org/current/doc/html/pcre2syntax.html#SEC4) say that `.` matching new line is not the default (opposite of Boost).

**Comment by hanickadot** [comment]
It took me a while, it will be released soon, it's currently implemented in branch `multiline` more info in #134

**Comment by bstaletic** [comment]
I've just given the multiline branch a try and it worked perfectly for my use case. Thanks!
